### PR TITLE
Creates interface to compare metadata file with directory of dig objects.

### PR DESCRIPTION
**Pull request by whunter**
**Creates interface to compare metadata file with directory of digital objects and produce report.**
* * *

**JIRA Ticket**: (https://webapps.es.vt.edu/jira/browse/LIBTD-2622) (:star:)

* Other Relevant Links (Meeting note, project page, related pull requests, etc.)

# What does this Pull Request do? (:star:)
Creates interface to compare metadata file with directory of digital objects and produce report.

# What's the changes? (:star:)
A in-depth description of the changes made by this PR. Technical details and possible side effects.

* Checks if metadata records are valid and contain required fields `Identifier`, `Title`
* Provides list of identifiers found in metadata file and the digital objects that are associated with them in the local directory
* Provides list of identifiers that DO NOT have any digital objects associated with them (likely in error)

# How should this be tested?

This PR is difficult to test. To test you will either need to:
* create a directory containing digital objects and a custom metadata csv that describes these objects 
* (or)
* download a partial collection from S3 that already includes a metadata csv that describes it. The problem is that this collection is **_really big_** (despite it being only a subset of the actual collection) It can be downloaded like this: `aws s3 cp s3://lee-ingest-testing/SWVA/BTR <YourLocalFolder> --recursive` 
* Then, visit `/siteAdmin/pre-ingest-check`
* Upload metadata csv (if you download the S3 collection it's `BTR.csv`
* Select the `BTR/` directory, click "Upload," and then confirm
* Check that the report is accurate

# Additional Notes:
Any additional information that you think would be helpful when reviewing this PR.

* branch: `LIBTD-2622`

# Interested parties
@yinlinchen 

(:star:) Required fields

### CODE REVIEW

File: src/pages/admin/ingestTools/PreIngestCheck.js [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    reader.readAsText(file);
+  };
+
+  handleDirInputChange = async event => {
+    let files = [];
+    let totalBytes = 0;
+    let totalFiles = 0;
+    for (const file of event.target.files) {
+      if (
+        ignoreFiles.indexOf(file.name) == -1 &&
```
Comment (andreaWaldren) [inline src/pages/admin/ingestTools/PreIngestCheck.js:67]: I'd probably change this to ===

File: src/css/adminForms.scss [hunk 2]
```diff
@@ -132,3 +132,47 @@ p.errorMsg {
...
+  }
+}
+#empty-identifiers {
+  .identifier-record {
+    color: red;
+    margin-bottom: 2px;
+  }
+}
+
+#invalid-records > .invalid-record {
```
Comment (andreaWaldren) [inline src/css/adminForms.scss:170]: Maybe add "display: block" so each one shows up on a new line?

**Review by andreaWaldren — CHANGES_REQUESTED** [review]
I think this should be ok to approve after the push to delete the unused cryptoJS package and possibly adding some sort of navigation from this form to the other parts of the admin site. Great work overall!

**Review by andreaWaldren — APPROVED** [review]

File: src/css/adminForms.scss (hunk 2 above)
Comment (whunter) [reply src/css/adminForms.scss:184]: 👍

**Pull request merged**